src/runtime/server/middleware/session/index.ts

```diff
@@ -136,7 +150,15 @@ const ensureSession = async (event: H3Event) => {
 
   let session = await getSession(event)
   if (!session) {
-    session = await newSession(event)
+    if (sessionOptions.saveUninitialized) {
+      session = await newSession(event)
+    } else {
+      // 1. Create an empty session object in the event context
+      event.context.session = {}
```

Comment:
This isn't a good idea, it is just a workaround so the rest of the module behaves as before. Initializing an empty session and only saving it when `saveUninitialized` is true is implemented in #48. I would suggest to merge #48 and not this PR and #47 individually.